src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,855 @@
+            }
+
+            byte[] trimmedReadBuffer = new byte[readBuffer.Length - 1];
+            Array.Copy(readBuffer, 1, trimmedReadBuffer, 0, trimmedReadBuffer.Length);
+            readBuffer = trimmedReadBuffer;
+
+            _registerCache[registerAddress] = ByteArrayToUInt32(readBuffer);
+
+            // Debug.WriteLine($"Read Register - address: {registerAddress.ToString("X2")}, command: {commandByte.ToString("X2")}, received: {String.Join(' ', readBuffer.Select(x => x.ToString("X2")))}");
+            return _registerCache[registerAddress];
+        }
+
+        /// <summary>
+        /// Writes data into a register
+        /// </summary>
+        protected void SetRegisterValue(Register register, uint registerValue)
+        {
+            byte registerAddress = (byte)register;
+            byte byteNumber = _registerSize[registerAddress];
+            byte[] writeBuffer = new byte[byteNumber + 1];
+
+            byte commandByte = (byte)((byte)CommunicationsRegisterBits.WriteOperation | GetCommAddress(registerAddress));
+            writeBuffer[0] = commandByte;
+
+            byte[] buffer = UInt32ToByteArray(registerValue, byteNumber);
+            Array.Copy(buffer, 0, writeBuffer, 1, byteNumber);
+
+            lock (_spiTransferLock)
+            {
+                _spiDevice.Write(writeBuffer);
+            }
+
+            byte[] trimmedWriteBuffer = new byte[writeBuffer.Length - 1];
+            Array.Copy(writeBuffer, 1, trimmedWriteBuffer, 0, trimmedWriteBuffer.Length);
+            writeBuffer = trimmedWriteBuffer;
+
+            // Debug.WriteLine($"Write Register - address: {registerAddress.ToString("X2")}, command: {commandByte.ToString("X2")}, sent: {String.Join(' ', writeBuffer.Select(x => x.ToString("X2")))}");
+        }
+
+        /// <summary>
+        /// Gets all the register values
+        /// </summary>
+        /// <returns>List of the values of the registers of the ADC</returns>
+        protected List<uint> GetAllRegisterValues()
+        {
+            List<uint> result = new List<uint>
+            {
+                GetRegisterValue(Register.Status),
+                GetRegisterValue(Register.Mode),
+                GetRegisterValue(Register.Configuration),
+                GetRegisterValue(Register.Data),
+                GetRegisterValue(Register.ID),
+                GetRegisterValue(Register.GPOCON),
+                GetRegisterValue(Register.Offset),
+                GetRegisterValue(Register.FullScale)
+            };
+
+            return result;
+        }
+
+        private static int GetCommAddress(int x)
+        {
+            return (((x) & 0x07) << 3);
+        }
+
+        /// <summary>
+        /// Returns the human-readable representation of a register
+        /// </summary>
+        /// <param name="register">The register you need the text-representation of</param>
+        /// <returns></returns>
+        public string RegisterToString(Register register)
```

Comment:
most of the devices made Register enum internal, while those are useful when debugging it's more confusing for users who didn't read datasheet

Reply:
The Register enum can't be internal because of the RegisterToString() method that can be used outside of he library. As you said it's useful for debugging, or in cases when the user needs more detailed information about the state of the device.

Reply:
I'd prefer we kept the debugging stuff for library implementers not for everyone

Reply:
I agree with @krwq here, basically we try device bindings to abstract as much as possible everything related to the actual communication between the device as well as the data-sheet contents, because the whole point of them is to have easy methods to use and only exposing what adds functionality of the sensor/device. Having public API like this will just make it harder to use.